index.ts

```diff
@@ -259,7 +252,7 @@ export default class Analytics {
     const hasReachedFlushAt = this.queue.length >= this.flushAt;
     const hasReachedQueueSize =
       this.queue.reduce((sum, item) => sum + JSON.stringify(item).length, 0) >=
-      this.maxInternalQueueSize;
+      this.maxQueueSizeInBytes;
```

Comment:
if we want to add some handling to shorten the flushed events to < 4mb we'd probably add it in here and it would also impact the runtime speed. I'm thinking something like build a map of event: eventSize and remove events from the upcoming flush until we are under `maxQueueSizeInBytes`. (those events would get included in the next flush)